README.md

```diff
@@ -8,16 +8,165 @@ Local.
 For more information, see [Getting started with Cloudant NoSQL DB](https://www.ng.bluemix.net/docs/services/Cloudant/index.html)
 
 The `loopback-connector-cloudant` module is the Cloudant connector for the Loopback framework.
+<!-- TOC -->
+
+- [loopback-connector-cloudant](#loopback-connector-cloudant)
+- [Design](#design)
+    - [No partial update](#no-partial-update)
+    - [No frequent modification](#no-frequent-modification)
+    - [Conflict control](#conflict-control)
+- [Model](#model)
+    - [Map between model and document](#map-between-model-and-document)
+    - [Model-specific configuration](#model-specific-configuration)
+    - [_rev property](#_rev-property)
+- [Installation](#installation)
+- [Config](#config)
+    - [generate datasource](#generate-datasource)
+    - [datasource config(`url` vs `username` & `password`)](#datasource-configurl-vs-username--password)
+    - [requests plugin](#requests-plugin)
+    - [example usage](#example-usage)
+- [CRUD](#crud)
+    - [update](#update)
+- [Migration](#migration)
+    - [autoupdate vs automigrate](#autoupdate-vs-automigrate)
+    - [isActual](#isactual)
+    - [example code](#example-code)
+- [Discovery](#discovery)
+- [Query](#query)
+- [Setup Cloudant instance](#setup-cloudant-instance)
+- [Testing](#testing)
+- [More Info](#more-info)
+- [Feature backlog](#feature-backlog)
+    - [View](#view)
+    - [Index](#index)
+
+<!-- /TOC -->
+
+# Design
```

Comment:
@jannyHou  per our discussion, I'm thinking of putting the `Design` and `Model` sections under `Getting Started`.  So the structure could be 
- Getting Started
    - Things to consider (the old `Design` section - we could think of a better name)
    - Best Practices (the old `Model` section)